lib/timescaledb/cleaner.ex

```diff
@@ -32,26 +33,34 @@ defmodule Membrane.RTC.Engine.TimescaleDB.Cleaner do
     apply(GenServer, function, [__MODULE__, reporter_options, gen_server_options])
   end
 
-  @impl true
-  def init(args) do
-    state = Map.new(args)
+  @spec stop(cleaner()) :: :ok
+  def stop(cleaner) do
+    GenServer.cast(cleaner, :stop)
+  end
 
-    %{cleanup_interval: cleanup_interval, metrics_lifetime: _metrics_lifetime, repo: _repo} =
-      state
+  @impl true
+  def init(opts) do
+    state = %{
+      repo: opts[:repo],
+      cleanup_interval_ms:
+        opts[:cleanup_interval] |> Membrane.Time.seconds() |> Membrane.Time.as_milliseconds(),
```

Comment:
If `cleanup_interval` is not set this will fail. You should either raise in `do_start` or provide a default here, e.g
```suggestion
        opts
        |> Map.get(:cleanup_interval, 60)
        |> Membrane.Time.seconds()
        |> Membrane.Time.as_milliseconds(),
```